owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/RemoteShare.kt

```diff
@@ -0,0 +1,203 @@
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class RemoteShare : Parcelable, Serializable {
+    var shareWith: String = ""
+    var path: String = ""
+    var token: String = ""
+    var sharedWithDisplayName: String = ""
+    var sharedWithAdditionalInfo: String = ""
+    var name: String = ""
+    var shareLink: String = ""
+    var fileSource: Long = 0
+    var itemSource: Long = 0
+    var shareType: ShareType? = null
+    var permissions: Int = 0
+    var sharedDate: Long = 0
+    var expirationDate: Long = 0
+    var isFolder: Boolean = false
+    var userId: Long = 0
+    var remoteId: Long = 0
+        private set
+
+    val isPasswordProtected: Boolean
+        get() = ShareType.PUBLIC_LINK == shareType && shareWith.isNotEmpty()
+
+    constructor() : super() {
+        resetData()
+    }
+
+    constructor(path: String?) {
+        resetData()
+        if (path == null || path.isEmpty() || !path.startsWith(FileUtils.PATH_SEPARATOR)) {
+            Log_OC.e(TAG, "Trying to create a RemoteShare with a non valid path")
+            throw IllegalArgumentException("Trying to create a RemoteShare with a non valid path: " + path!!)
+        }
+        this.path = path
+    }
+
+    /**
+     * Used internally. Reset all file properties
+     */
+    private fun resetData() {
+        shareWith = ""
+        path = ""
+        token = ""
+        sharedWithDisplayName = ""
+        sharedWithAdditionalInfo = ""
+        name = ""
+        shareLink = ""
+        fileSource = 0
+        itemSource = 0
+        shareType = ShareType.NO_SHARED
+        permissions = DEFAULT_PERMISSION
+        sharedDate = 0
+        expirationDate = 0
+        sharedWithAdditionalInfo = ""
+        isFolder = false
+        userId = -1
+        remoteId = -1
+    }
+
+    fun setIdRemoteShared(remoteId: Long) {
+        this.remoteId = remoteId
+    }
+
+    /**
+     * Reconstruct from parcel
+     *
+     * @param source The source parcel
+     */
+    protected constructor(source: Parcel) {
+        readFromParcel(source)
+    }
+
+    fun readFromParcel(source: Parcel) {
+        shareWith = source.readString()
+        path = source.readString()
+        token = source.readString()
+        sharedWithDisplayName = source.readString()
+        sharedWithAdditionalInfo = source.readString()
+        name = source.readString()
+        shareLink = source.readString()
+        fileSource = source.readLong()
+        itemSource = source.readLong()
+        try {
+            shareType = ShareType.valueOf(source.readString())
+        } catch (x: IllegalArgumentException) {
+            shareType = ShareType.NO_SHARED
+        }
+        permissions = source.readInt()
+        sharedDate = source.readLong()
+        expirationDate = source.readLong()
+        isFolder = source.readInt() == 0
+        userId = source.readLong()
+        remoteId = source.readLong()
+    }
+
+    override fun describeContents(): Int {
```

Comment:
`override fun describeContents(): Int = this.hashCode()`

Reply:
`override fun describeContents() = this.hashCode()`
and on a lot of places you can remove the type too